### PR TITLE
Backend - Maintenance Issues CRUD

### LINKED ISSUE #138: Create  GET endpoints for retrieving all Maintenance Issues and retrieving a Maintenance Issue by Id
**Is your feature request related to a problem? Please describe.**

Currently, there is no implementation on logging maintenance issues for the platform.

**Describe the solution you'd like**

As part of implementing the logging maintenance issue feature, when the GET endpoints are called, it will return the necessary Maintenance Issue(s).

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #140: Create POST Maintenance Issue Endpoint
**Is your feature request related to a problem? Please describe.**

Need to add a POST endpoint that will enable users to post their maintenance issues to MongoDB database. 

**Describe the solution you'd like**

The POST maintenance endpoint will retrieve a Maintenance Issue object from the client and add it to the issues database. 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #145: Create Maintenance Issue Model
**Is your feature request related to a problem? Please describe.**

I need given data to conform to standards when uploaded to the database.

**Describe the solution you'd like**

A nice model schema for data to follow.

**Describe alternatives you've considered**

Make a single typescript interface, but it is less flexible.

**Additional context**

n/a

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #148: Create PUT issue endpoint
**Is your feature request related to a problem? Please describe.**
The current backend does not support updating an issue information such as adding extra details and changing resolved status

**Describe the solution you'd like**
The PUT issue endpoint will update the Maintenance Issue data with the request body data. 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #149: Create DELETE issue endpoint
**Is your feature request related to a problem? Please describe.**

If an issue is no longer relevant or wasn't an issue or I want to hide it from my landlord I need to get rid of it!

**Describe the solution you'd like**

I would like the issue to be deleted from the database so it cannot be shown ever again. I want to use mongoose to delete the record with a particular id. 

**Describe alternatives you've considered**

No real alternative here.

**Additional context**

n/a

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### PR DESCRIPTION
**Pull request by butterplup**
# Description

Added a model/schema for the issues as well as the basic CRUD services. This is so we can work on the controllers and have a standard model design to work off. PR is dependent on mongoose and nest.js but no other areas of code.

Fixes/resolves #138, #140, #145, #148, #149

## Screenshots

n/a

## Type of change

- [x] **New feature** (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

I have completed these steps when making this pull request:

- [x] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [x] I have labelled the PR appropriately
- [x] I have assigned myself to the PR

Before opening the PR for review:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

File: packages/backend/src/db/issue/issue.schema.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description: string;
+
+  @Prop({ type: Types.ObjectId, ref: 'House' })
+  house: Types.ObjectId;
+
+  @Prop({ type: Types.ObjectId, ref: 'User' })
+  logger: Types.ObjectId;
+
+  @Prop()
+  loggedDate: Date;
```
Comment (lanxdrey19) [inline packages/backend/src/db/issue/issue.schema.ts:21]: Everything looks great! I just had a minor question about this part. The date will be automatically created right? so does it need to have some default value or is it already done automatically?

**Review by lanxdrey19 — COMMENTED** [review]
The structure looks great. I just had a minor question.  The date will be automatically created right? so does the date need to have some default value in the schema or is it already done automatically?

File: packages/backend/src/db/issue/issue.schema.ts (hunk 1 above)
Comment (butterplup) [reply packages/backend/src/db/issue/issue.schema.ts:21]: Yep, I just added a default value now 👍

**Review by lanxdrey19 — DISMISSED** [review]
LGTM!

**Comment by butterplup** [comment]
Make sure not to approve this until after the CRUD controller stuff is pushed.

**Comment by butterplup** [comment]
Peer coded PR with Sherman, Joel, Lance.

**Review by lanxdrey19 — APPROVED** [review]
LGTM

**Review by schi314 — APPROVED** [review]
LGTM!

**Pull request merged**